omas/omas_plot.py

```diff
@@ -446,3 +446,136 @@ def core_profiles_pressures(ods, time_index=0, ax=None, **kw):
     ax.set_xlim([0, 1])
     ax.legend(loc=0).draggable(True)
     return ax
+
+
+@add_to__ODS__
+def overlay(ods, ax=None, **kw):
+    """
+    Plots overlays of hardware/diagnostic locations on a tokamak cross section plot
+    :param ods: OMAS ODS instance
+    :param ax: Axes instance or None
+    :param kw: select plots by setting their names to True; e.g.: if you want the gas plot, set gas=True as a keyword.
+        Instead of True to simply turn on an overlay, you can pass a dict of keywords to pass to a particular overlay
+        method, as in thomson={'labelevery': 5}
+    """
+    if ax is None:
+        ax = pyplot.gca()
+    if kw.pop('gas', False):
+        gas_overlay(ods, ax)
+    if kw.get('thomson', False):
+        tskw = kw.get('thomson', {}) if isinstance(kw.get('thomson', {}), dict) else {}
+        thomson_overlay(ods, ax, **tskw)
+    if kw.get('bolometer', False):
+        bolkw = kw.get('bolometer', {}) if isinstance(kw.get('bolometer', {}), dict) else {}
+        bolometer_overlay(ods, ax, **bolkw)
```

Comment:
I suspect this function could be generalized to process `**kw`, such that the keys of **kw are the names of the IDS's, like
![image](https://user-images.githubusercontent.com/782050/41053350-2163e52c-6970-11e8-9ec5-7e34a75010dd.png)
(with `name.replace(' ','_')`) and you can use something like
```python
for k,v in kw.items():
  k = k.replace(' ','_')
  try:
    eval(k+'_overlay)
  except NameError:
    print('%s does not exist'%(k+'_overlay'))
    print('Must be one of:')
    overlays = [d for d in dir() if d.endswith('_overlay')]
    print(overlays)
    # raise NotImplementedError ?
  else:
    processed_kw = v if isinstance(v,dict) else {}
    eval(k+'_overlay')(**processed_kw)
```

Reply:
You are right.